### DISCUSSION TITLE
send variable to another function in the same component

### DISCUSSION THREAD

**Discussion by jererosas** [livewire/livewire#5708] [1. Help]
I want to know if variables can be passed to other livewire functions in the same component
I am currently listening for the id number of a room

Button emit:
<button  data-target="#registerModal{{ $habitacion->id }}" wire:click="showModal({{$room->id}})">

Component to emit:

```
  public function ShowModal($id)
    {


        $this->emit('ShowModal', $id);

    }
```

```
public $room;
protected $listeners = ['ShowModal' => 'readRoom'];

public function readRoom($id)
{
   $this->room = $id; // room number to pass to AddTocCart function //

}

public function addToCart($id)
    {


        

        $product = Product::find($id);


        \Cart::add([
            'id' => $product->id,
            'name' => $product->nombre,
            'price' => $product->precio,
            'quantity' => 1,
            'attributes' => array(
                'image' => $product->imagen,
                 'habitacion' => $this->room,  // receive here room number //
            )
        ]);

    }
```

I did a dd($this->room) but I get NULL

**How can I pass the $room variable to the addToCart function?**

**Comment by joshhanley** [comment]
@jererosas i don’t follow what you’re trying to do? Your referencing `$this->room` yet I don’t see that property in the code you’ve shared. And I don’t see how your calling the function. Please share all relevant code and make sure we can follow it, so we can assist better 🙂

**Reply by jererosas** [reply]
I'm sorry, edit my post with the complete code and comment in it what I need
